plugins/inputs/gnmi/gnmi.go

```diff
@@ -272,8 +273,13 @@ func (c *GNMI) Start(acc telegraf.Accumulator) error {
 		go func(addr string) {
 			defer c.wg.Done()
 
+			host, _, err := net.SplitHostPort(addr)
+			if err != nil {
+				c.Log.Errorf("unable to parse address %s: %v", addr, err)
+			}
 			h := handler{
 				address:             addr,
+				host:                host,
```

Comment:
Do we need the `address` in the handler then?

Reply:
No if I add port also to handler, is that better? To drop address and add host + port, as `client, err := grpc.NewClient(h.address, opts...)` still needs host + port

Reply:
dropped address and added port to handler. Changed initial value to be 1, and report 0 if connecting fails. First gather when starting telegraf will report 0 for all, so probably better to report 1 and change to 0 if fails